File: 415.字符串相加.java
/*
 * @lc app=leetcode.cn id=415 lang=java
 *
 * [415] 字符串相加
 */

// @lc code=start
class Solution {
    public String addStrings(String num1, String num2) {
        StringBuilder sb = new StringBuilder();
        int i = num1.length()-1;
        int j = num2.length()-1;
        int carry = 0;
        int tmp, n1, n2;
        while (i >= 0 || j >= 0) {
            // n1 = i >= 0 ? Integer.parseInt(String.valueOf(num1.charAt(i))) : 0;
            n1 = i >= 0 ? num1.charAt(i)-48 : 0;
            // n2 = j >= 0 ? Integer.parseInt(String.valueOf(num2.charAt(j))) : 0;
            n2 = j >= 0 ? num2.charAt(j)-48 : 0;
            tmp = n1 + n2 + carry;
            carry = tmp / 10;
            sb.append(tmp % 10);
            i--;
            j--;
        }
        if (carry == 1)
            sb.append(1);
        return sb.reverse().toString();

    }
}
// @lc code=end

File: 415.字符串相加.py
#
# @lc app=leetcode.cn id=415 lang=python3
#
# [415] 字符串相加
#

# @lc code=start


class Solution:
    def addStrings(self, num1: str, num2: str) -> str:

        if len(num1) < len(num2):
            num1, num2 = num2, num1
        carry = 0
        res = ''
        for i in range(len(num1)):
            n1 = int(num1[len(num1)-i-1])
            n2 = int(num2[len(num2)-i-1]) if i < len(num2) else 0
            carry, tmp = divmod(n1+n2+carry, 10)
            res = str(tmp) + res
        if carry == 1:
            res = '1'+res
        return res


# @lc code=end
